light-zk.js/src/utxo.ts

```diff
@@ -219,14 +219,34 @@ export class Utxo {
     this.includeAppData = includeAppData;
     this.transactionVersion = "0";
 
+    if (
+      assets[1].toBase58() === SystemProgram.programId.toBase58() &&
+      !amounts[1].isZero()
+    ) {
+      throw new UtxoError(
+        UtxoErrorCode.POSITIVE_AMOUNT,
+        "constructor",
+        `spl amount cannot be positive, amounts[1] = ${
+          amounts[1] ?? "undefined"
+        }`,
+      );
+    }
     // TODO: make variable length
-    if (assets[1].toBase58() != SystemProgram.programId.toBase58()) {
+    else if (assets[1].toBase58() != SystemProgram.programId.toBase58()) {
       this.assetsCircuit = [
         hashAndTruncateToCircuit(SystemProgram.programId.toBytes()),
         hashAndTruncateToCircuit(this.assets[1].toBytes()),
       ];
-    } else if (this.amounts[0].toString() === "0") {
+    } else if (this.amounts[0].isZero()) {
       this.assetsCircuit = [BN_0, BN_0];
+    }
+    // @matteo: adding condition for amounts =! 0
+    else if (!this.amounts[0].isZero()) {
```

Comment:
Why do you always throw an error if the amounts[0] is non zero?

amounts[0] is the sol amount of the utxo it's intended to be non zero when used and is allowed to be zero when it is used as a filling utxo.

Reply:
made no sense, I commented it

Reply:
top, tests look good can you delete it?
Or is there a reason why we would want to keep the comment?

Reply:
removed!